### PR TITLE
Apply other check-sources plugins

### PR DESCRIPTION
**Pull request by arogge**
This PR updates the codebase to comply with our automatic guidelines:
* clang-format applied to all C/C++ sources
* cmake-format applied to all CMake sources
* Python code reformatted with black
* trim trailing whitespace and excess newlines at EOF
* use `#!/usr/bin/env python` as shebang for python
* remove executable flags from non-executable files

### CODE REVIEW

File: core/src/dird/jobq.cc [hunk 1]
```diff
@@ -974,8 +972,8 @@ void DecReadStore(JobControlRecord* jcr)
                ->NumConcurrentReadJobs);
     }
 
-    if (jcr->impl->res.read_storage->runtime_storage_status
-            ->NumConcurrentJobs < 0) {
+    if (jcr->impl->res.read_storage->runtime_storage_status->NumConcurrentJobs <
```
Comment (franku) [inline core/src/dird/jobq.cc:975]: Could we wrap the "<" to the next line?

**Review by franku — COMMENTED** [review]
Good work!

File: core/cmake/BareosGenerateDebianInfo.cmake [hunk 2]
```diff
@@ -20,65 +20,89 @@
...
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-python-plugins-common
-      DEBIAN_CONTROL_STORAGEDAEMON_PYTHON_PLUGINS_COMMON)
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-director-python-plugins-common
-      DEBIAN_CONTROL_DIRECTOR_PYTHON_PLUGINS_COMMON)
+    file(
+      READ
+      ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon-python-plugins-common
+      DEBIAN_CONTROL_FILEDAEMON_PYTHON_PLUGINS_COMMON
+    )
+    file(READ
```
[the remark this answers is not in the document]
Comment (arogge) [reply core/cmake/BareosGenerateDebianInfo.cmake:82]: that line exceeds the length limit and is therefore wrapped

File: core/src/stored/backends/generic_tape_device.cc [hunk 3]
```diff
@@ -154,7 +155,8 @@ void generic_tape_device::OpenDevice(DeviceControlRecord* dcr, DeviceMode omode)
  */
 bool generic_tape_device::eod(DeviceControlRecord* dcr)
 {
-  struct mtop mt_com{};
+  struct mtop mt_com {
```
[the remark this answers is not in the document]
Comment (arogge) [reply core/src/stored/backends/generic_tape_device.cc:158]: i looked at the clang-format documentation (twice!) and there is no way to configure this.
So to fix it, we will have to change `struct mtop mt_com{};` to `mtop mt_com{};` (which is identical).
Other variants (like `mtop mt_com;` or `struct mtop mt_com = 0`) produce different assembler-code, which may be an issue.

**Pull request merged**